From 9bfa6a3d05584eecd6772f07db2d441fc3a0b524 Mon Sep 17 00:00:00 2001
From: Hu Kejun <william.hu@rock-chips.com>
Date: Thu, 21 Feb 2019 08:41:30 +0800
Subject: [PATCH] arm64: dts: rockchip: add vm149c for ov13850

 Conflicts:
	arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-edp.dtsi
	arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux-for-rk1808-cascade.dts
	arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10-linux.dts
	arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11-linux.dts

remove the old camera module information.

Change-Id: Ida81f26ebc718840d750919a6e591da6d5865765
Signed-off-by: Hu Kejun <william.hu@rock-chips.com>
---
 .../dts/rockchip/rk3399-sapphire-excavator-edp.dtsi    | 10 ++++++++++
 ...399-sapphire-excavator-linux-for-rk1808-cascade.dts |  9 +++++++++
 .../dts/rockchip/rk3399-sapphire-excavator-linux.dts   | 10 ++++++++++
 .../boot/dts/rockchip/rk3399pro-evb-v10-linux.dts      |  9 +++++++++
 .../boot/dts/rockchip/rk3399pro-evb-v11-linux.dts      | 10 ++++++++++
 5 files changed, 48 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-edp.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-edp.dtsi
index 85957178dfb6..118121fb2cd8 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-edp.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-edp.dtsi
@@ -245,6 +245,14 @@
 		};
 	};
 
+	vm149c: vm149c@0c {
+		compatible = "silicon touch,vm149c";
+		status = "okay";
+		reg = <0x0c>;
+		rockchip,camera-module-index = <0>;
+		rockchip,camera-module-facing = "back";
+	};
+
 	ov13850: ov13850@10 {
 		compatible = "ovti,ov13850";
 		status = "okay";
@@ -259,12 +267,14 @@
 		pwdn-gpios = <&gpio1 4 GPIO_ACTIVE_HIGH>;
 		pinctrl-names = "rockchip,camera_default";
 		pinctrl-0 = <&cif_clkout>;
+
 		rockchip,camera-module-index = <0>;
 		rockchip,camera-module-facing = "back";
 		rockchip,camera-module-name = "CMK-CT0116";
 		rockchip,camera-module-lens-name = "Largan-50013A1";
 		lens-focus = <&vm149c>;
 		flash-leds = <&sgm3784_led0 &sgm3784_led1>;
+
 		port {
 			ucam_out0: endpoint {
 				remote-endpoint = <&mipi_in_ucam0>;
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux-for-rk1808-cascade.dts b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux-for-rk1808-cascade.dts
index 7e93b9e05c88..bc143d19bab4 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux-for-rk1808-cascade.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux-for-rk1808-cascade.dts
@@ -400,6 +400,14 @@
 		};
 	};
 
+	vm149c: vm149c@0c {
+		compatible = "silicon touch,vm149c";
+		status = "okay";
+		reg = <0x0c>;
+		rockchip,camera-module-index = <0>;
+		rockchip,camera-module-facing = "back";
+	};
+
 	ov13850: ov13850@10 {
 		compatible = "ovti,ov13850";
 		status = "okay";
@@ -412,6 +420,7 @@
 		pwdn-gpios = <&gpio1 4 GPIO_ACTIVE_HIGH>;
 		pinctrl-names = "rockchip,camera_default";
 		pinctrl-0 = <&cif_clkout>;
+		lens-focus = <&vm149c>;
 
 		port {
 			ucam_out0: endpoint {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts
index 892fef952753..ed059efcc576 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-linux.dts
@@ -360,6 +360,14 @@
 		};
 	};
 
+	vm149c: vm149c@0c {
+		compatible = "silicon touch,vm149c";
+		status = "okay";
+		reg = <0x0c>;
+		rockchip,camera-module-index = <0>;
+		rockchip,camera-module-facing = "back";
+	};
+
 	ov13850: ov13850@10 {
 		compatible = "ovti,ov13850";
 		status = "okay";
@@ -373,6 +381,8 @@
 		pinctrl-names = "rockchip,camera_default";
 		pinctrl-0 = <&cif_clkout>;
 
+		lens-focus = <&vm149c>;
+
 		port {
 			ucam_out0: endpoint {
 				remote-endpoint = <&mipi_in_ucam0>;
diff --git a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10-linux.dts b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10-linux.dts
index accbded9ce77..c43a84428136 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10-linux.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v10-linux.dts
@@ -706,6 +706,14 @@
 		layout = <3>;
 	};
 
+	vm149c: vm149c@0c {
+		compatible = "silicon touch,vm149c";
+		status = "okay";
+		reg = <0x0c>;
+		rockchip,camera-module-index = <0>;
+		rockchip,camera-module-facing = "back";
+	};
+
 	ov13850: ov13850@10 {
 		compatible = "ovti,ov13850";
 		status = "disabled";
@@ -718,6 +726,7 @@
 		pwdn-gpios = <&gpio4 25 GPIO_ACTIVE_HIGH>;
 		pinctrl-names = "rockchip,camera_default";
 		pinctrl-0 = <&cif_clkout>;
+		lens-focus = <&vm149c>;
 
 		port {
 			ucam_out0: endpoint {
diff --git a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11-linux.dts b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11-linux.dts
index 96b18a20f8da..473721804a01 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11-linux.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399pro-evb-v11-linux.dts
@@ -716,6 +716,14 @@
 		layout = <3>;
 	};
 
+	vm149c: vm149c@0c {
+		compatible = "silicon touch,vm149c";
+		status = "okay";
+		reg = <0x0c>;
+		rockchip,camera-module-index = <0>;
+		rockchip,camera-module-facing = "back";
+	};
+
 	ov13850: ov13850@10 {
 		compatible = "ovti,ov13850";
 		status = "okay";
@@ -729,6 +737,8 @@
 		pinctrl-names = "rockchip,camera_default";
 		pinctrl-0 = <&cif_clkout>;
 
+		lens-focus = <&vm149c>;
+
 		port {
 			ucam_out0: endpoint {
 				remote-endpoint = <&mipi_in_ucam0>;
-- 
2.35.3

